src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -21,99 +19,1454 @@ const ActorInit En_Mb_InitVars = {
+        phi_a2 = -0x4000;
+    } else {
+        if (this->actor.posRot.rot.y >= 0x20E1) {
+            phi_a2 = 0x4000;
+        }
+    }
+    cosAngle = Math_Coss(phi_a2);
+    sinAngle = Math_Sins(phi_a2);
+    cosAngle = (cosAngle >= 0) ? cosAngle : -cosAngle;
+    sinAngle = (sinAngle >= 0) ? sinAngle : -sinAngle;
+    xDist = this->actor.posRot.pos.x - player->actor.posRot.pos.x;
+    zDist = this->actor.posRot.pos.z - player->actor.posRot.pos.z;
+    xDistABS = (xDist >= 0) ? xDist : -xDist;
+    if (xDistABS < cosAngle * 37.0f + sinAngle * 400.0f) {
+        if (0 <= zDist) {
+            zDistABS = zDist;
+        } else {
+            zDistABS = -zDist;
+        }
+        if (zDistABS < (sinAngle * 37.0f) + (cosAngle * 400.0f)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void func_80AA66A0(EnMb* this, GlobalContext* globalCtx) {
+    Path* path;
+    s16 yawToPoints;
+    Vec3f points;
+    Vec3s* pointPath;
+    s16 yawDiff;
+    s32 pathCount;
+    s32 pathIndex;
+    s32 yawDiffABS;
+
+    path = &globalCtx->setupPathList[this->unk_35D];
+    for (pathIndex = 0, pathCount = path->count - 1; pathCount >= 0; pathCount--, pathIndex++) {
+        pointPath = &((Vec3s*)SEGMENTED_TO_VIRTUAL(path->points))[pathIndex];
+        points.x = pointPath->x;
+        points.y = pointPath->y;
+        points.z = pointPath->z;
+        yawToPoints = Math_Vec3f_Yaw(&this->actor.posRot.pos, &points);
+        yawDiff = (yawToPoints) - this->actor.yawTowardsLink;
+        yawDiffABS = (yawDiff >= 0) ? yawDiff : -yawDiff;
+        if (yawDiffABS <= 0x1770) {
+            this->actor.posRot.rot.y = yawToPoints;
+            if (pathIndex == this->unk_35C) {
+                this->unk_35E = -this->unk_35E;
+            }
+            this->unk_344 = points;
+            this->unk_35C = pathIndex;
+            return;
+        }
+    }
+}
+
+void func_80AA6830(EnMb* this) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_060041A8, -4.0f);
+    this->actor.speedXZ = 0.0f;
+    this->unk_32A = Math_Rand_S16Offset(30, 50);
+    this->unk_320 = 6;
+    EnMb_SetupAction(this, func_80AA7134);
+}
+
+void func_80AA6898(EnMb* this) {
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_0600EBE4);
+    this->actor.speedXZ = 0.0f;
+    this->unk_32A = Math_Rand_S16Offset(30, 50);
+    this->unk_320 = 6;
+    EnMb_SetupAction(this, func_80AA8DD8);
+}
+
+void func_80AA68FC(EnMb* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_060041A8, -4.0f);
+    this->actor.speedXZ = 0.0f;
+    this->unk_32A = Math_Rand_S16Offset(40, 80);
+    this->unk_320 = 6;
+    func_80AA6444(this, globalCtx);
+    EnMb_SetupAction(this, func_80AA71AC);
+}
+
+void func_80AA6974(EnMb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06009FC0, 0.0f, 0.0f, SkelAnime_GetFrameCount(&D_06009FC0), 0, -4.0f);
+    this->actor.speedXZ = 0.59999996f;
+    this->unk_32A = Math_Rand_S16Offset(50, 70);
```

Comment:
can that be 0.60f?

Reply:
It can not.  Probably some kind of calculation.